dev/tests/unit/Magento/FunctionalTestFramework/Test/Objects/ActionGroupObjectTest.php

```diff
@@ -261,6 +261,72 @@ public function testExceptionOnMissingArguments()
+                new ActionObject('updateDataStepKey', 'updateData', ['selector' => 'value']),
+                new ActionObject('createDataStepKey', 'createData', ['selector' => 'value']),
+                new ActionObject('grabAttributeFromStepKey', 'grabAttributeFrom', ['selector' => 'value']),
+                new ActionObject('grabCookieStepKey', 'grabCookie', ['selector' => 'value']),
+                new ActionObject('grabFromCurrentUrlStepKey', 'grabFromCurrentUrl', ['selector' => 'value']),
+                new ActionObject('grabMultipleStepKey', 'grabMultiple', ['selector' => 'value']),
+                new ActionObject('grabPageSourceStepKey', 'grabPageSource', ['selector' => 'value']),
+                new ActionObject('grabTextFromStepKey', 'grabTextFrom', ['selector' => 'value']),
+                new ActionObject('grabValueFromStepKey', 'grabValueFrom', ['selector' => 'value'])
+            ])
+            ->build();
+
+        $result = $actionGroupUnderTest->extractStepKeys();
+
+        $this->assertEquals('executeJSStepKey', $result[0]);
+        $this->assertEquals('magentoCLIStepKey', $result[1]);
+        $this->assertEquals('generateDateStepKey', $result[2]);
+        $this->assertEquals('formatMoneyStepKey', $result[3]);
+        $this->assertEquals('deleteDataStepKey', $result[4]);
+        $this->assertEquals('getDataStepKey', $result[5]);
+        $this->assertEquals('updateDataStepKey', $result[6]);
+        $this->assertEquals('createDataStepKey', $result[7]);
+        $this->assertEquals('grabAttributeFromStepKey', $result[8]);
+        $this->assertEquals('grabCookieStepKey', $result[9]);
+        $this->assertEquals('grabFromCurrentUrlStepKey', $result[10]);
+        $this->assertEquals('grabMultipleStepKey', $result[11]);
+        $this->assertEquals('grabPageSourceStepKey', $result[12]);
+        $this->assertEquals('grabTextFromStepKey', $result[13]);
+        $this->assertEquals('grabValueFromStepKey', $result[14]);
+    }
+
+    /**
+     * Tests the stepKey replacement with "stepKey + invocationKey" process filter
+     * Specific to actions that make are removed by a "require stepKey replacement" filter
+     */
+    public function testStepKeyReplacementFilteredOut()
+    {
+        $actionGroupUnderTest = (new ActionGroupObjectBuilder())
+            ->withActionObjects([
+                new ActionObject('clickStepKey', 'click', ['selector' => 'value']),
+                new ActionObject('conditionalClickStepKey', 'conditionalClick', ['selector' => 'value']),
+                new ActionObject('fillFieldStepKey', 'fillField', ['selector' => 'value'])
+            ])
+            ->build();
+
+        $result = $actionGroupUnderTest->extractStepKeys();
+
+        $this->assertNotContains('clickStepKey', $result);
+        $this->assertNotContains('conditionalClickStepKey', $result);
+        $this->assertNotContains('fillFieldStepKey', $result);
```

Comment:
Talked with Alex Calandra about this, and I think we should take this unit test in a slightly different direction:
* instead of building actions for every constant, pick one or two from `ActionGroupObject::YOUR_CONSTANT` and build action Objects using them.
* Then assert that the array contains the action types used, as opposed to index specific matching. Also assert on the size of `$result`.

Additionally, in both tests, try to make use of function-scoped variables instead of hardcoded strings like:
`new ActionObject($actionStepKey, $actionType, ['selector' => 'value']),`